### PR TITLE
`HeteroData` support for `T.ToSparseTensor`

### PR DESCRIPTION
**Pull request by rusty1s**

### CODE REVIEW

File: torch_geometric/transforms/to_sparse_tensor.py [hunk 1]
```diff
@@ -26,34 +31,40 @@ def __init__(self, remove_edge_index: bool = True,
...
+        for store in data.edge_stores:
+            if 'edge_index' not in store:
+                continue
+
+            row, col = store.edge_index
+            perm = (col * store.size(0)).add_(row).argsort()
+            row, col = row[perm], col[perm]
+
+            value = None
+            for key in store.keys('edge_weight', 'edge_attr', 'edge_type'):
```
Comment (JiaxuanYou) [inline torch_geometric/transforms/to_sparse_tensor.py:44]: Is this condition too ad-hoc? How about apply this permutation as long as `edge` appears in the attribute? I remember in OGB `edge_feat` is used, and in DeepSNAP/GraphGym we used `edge_feature`.

File: torch_geometric/transforms/to_sparse_tensor.py [hunk 2]
```diff
@@ -26,34 +31,40 @@ def __init__(self, remove_edge_index: bool = True,
...
+                if isinstance(store[key], Tensor):
+                    value = store[key][perm]
+                    if self.remove_edge_index:
```
Comment (JiaxuanYou) [inline torch_geometric/transforms/to_sparse_tensor.py:47]: The argument says `remove_edge_index`, but here is seems that `'edge_weight', 'edge_attr', 'edge_type'` are being removed?

File: torch_geometric/transforms/to_sparse_tensor.py (hunk 1 above)
Comment (rusty1s) [reply torch_geometric/transforms/to_sparse_tensor.py:44]: It is applied to any feature in `EdgeStorage`, but we add the first existing attribute of either `edge_weight`, `edge_attr` or `edge_type` as direct value to `SparseTensor`.

File: torch_geometric/transforms/to_sparse_tensor.py (hunk 2 above)
Comment (rusty1s) [reply torch_geometric/transforms/to_sparse_tensor.py:47]: Both `edge_index` and the first existing attribute of `edge_weight`, `edge_attr` or `edge_type` are removed (as it is now an attribute of `SparseTensor`.

**Comment by rusty1s** [comment]
I added an argument `attr` that denotes which attribute to add as a value to the `SparseTensor` object. By default, this is the `edge_weight` attribute to simplify applying sparse-matrix multiplication in weighted graphs.

**Pull request merged**